ethicml/utility/data_structures.py

```diff
@@ -256,6 +287,150 @@ def from_npz(cls, data_path: Path) -> DataTuple:
             )
 
 
+@dataclass
+class LabelTuple:
+    """A tuple of dataframes for the features, the sensitive attribute and the class labels."""
+
+    __slots__ = ("data", "s_column", "y_column", "name")
+    data: pd.DataFrame
+    s_column: str
+    y_column: str
+    name: str | None
+
+    def __post_init__(self) -> None:
+        assert self.s_column in self.data.columns, f"column {self.s_column} not present"
+        assert self.y_column in self.data.columns, f"column {self.y_column} not present"
+
+    @classmethod
+    def from_df(
+        cls, *, s: pd.Series[int], y: pd.Series[int], name: Optional[str] = None
+    ) -> LabelTuple:
+        """Make a LabelTuple."""
+        s_column = s.name
+        y_column = y.name
+        assert isinstance(s_column, str) and isinstance(y_column, str)
+        assert len(s) == len(y), "data has to have the same length"
+        return cls(
+            data=pd.concat([s, y], axis="columns", sort=False),  # type: ignore[arg-type]
+            s_column=s_column,
+            y_column=y_column,
+            name=name,
+        )
+
+    @classmethod
+    def from_np(cls, *, s: npt.NDArray, y: npt.NDArray, s_name: str, y_name: str) -> LabelTuple:
```

Comment:
Maybe make "s" and "y" the default names?

Reply:
sure, good idea